src/app/core/services/user.service.ts

```diff
@@ -0,0 +1,29 @@
+import { Injectable } from '@angular/core';
+import { AngularFirestore } from 'angularfire2/firestore';
+import { AngularFireAuth } from 'angularfire2/auth';
+import * as firebase from 'firebase/app';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class UserService {
+
+  constructor(
+    public db: AngularFirestore,
+    public afAuth: AngularFireAuth
+  ) { }
+
+
+  getCurrentUser() {
+    return new Promise<any>((resolve, reject) => {
+      const userStatus = firebase.auth().onAuthStateChanged(function(user) {
+        if (user) {
+          resolve(user);
+        } else {
+          reject('No user logged in');
+        }
+      });
+    });
```

Comment:
Can be shortened to 
`const userStatus = firebase.auth().onAuthStateChanged(user => user ? resolve(user) : reject('No user logged in')`